cleo_plugins/Audio/CSoundSystem.cpp

```diff
@@ -52,7 +53,7 @@ namespace CLEO
         if (initialized)
         {
             TRACE("Freeing BASS library");
-            BASS_Free();
+            std::thread(BASS_Free);
```

Comment:
why is there a deadlock?

Reply:
The library internally waits to some (COM?) objects to be finalized. BASS author says free can not be called in dll main. I tried to do it from some other callback like on game closing, but with same results. Other author suggestion is not to call free at all and let it be released automatically, but that makes game crashing (I guess due all ModLoader hacks).

Calling it from another thread works. It is visible in log that CLEO proceeded with other cleanup steps, game aslo closes without problems.

Reply:
Interesting. I would also remove free and let OS handle the unloading.

Is the crash reproducible 100%? Does it depend on other plugins installed in game?

Reply:
Seems like all you need is ModLoader

Reply:
Why it wasn't an issue in CLEO4?

Reply:
I have no idea. ModLoader is thin ice. In CLEO4 Bass was managed by core.

Reply:
Hm seems to be working without calling free at all. I have no idea why game was crashing for me when testing it previously.